election-server/src/util.rs

```diff
@@ -17,3 +28,83 @@ pub struct BallotSubmission {
     /// Whether the ballot proof could be verified.
     pub verified:         bool,
 }
+
+pub enum ElectionContractMarker {}
+
+pub type ElectionContract = ContractClient<ElectionContractMarker>;
+
+/// Verify the digest of `file` matches the expected `checksum`.
+pub fn verify_checksum(file: &Path, expected_checksum: [u8; 32]) -> anyhow::Result<()> {
+    use sha2::Digest;
+    let computed_hash: [u8; 32] = sha2::Sha256::digest(
+        std::fs::read(file)
+            .with_context(|| format!("Could not digest file at location: {:?}", file))?,
+    )
+    .into();
```

Comment:
```suggestion
    let mut hasher = sha2::Sha256::new();
    let source = std::fs::File::open(file).context("Unable to open file.")?;
    std::io::copy(&mut source, &mut hasher).with_context(|| format!("Could not digest file at location: {:?}", file))?;
    hasher.finalize().into()
```

THis is to avoid loading the contents of the entire file into memory. It can be quite large.